CONTRIBUTING.md

```diff
@@ -1,26 +1,137 @@
+[commit_types]: https://github.com/commitizen/conventional-commit-types/blob/master/index.json
+[conventional_commits]: https://www.conventionalcommits.org/en/v1.0.0/
+
+### Formatting.
+
+[Prettier](https://prettier.io/) is used to ensure consistent code formatting in this repository.
+This is normally transparent as it automatically activated in a pre-commit hook using [lint-staged](https://github.com/okonet/lint-staged).
+However, this does mean that dev flows that do not use a full dev env (e.g editing directly on github)
+may result in voter failures due to formatting errors.
+
+### Compiling
+
+Use the following npm scripts at the repo's **root** to compile **all** the TypeScript sub-packages.
+
+- `yarn compile`
+- `yarn compile:watch` (will watch files for changes and re-compile as needed)
+
+These scripts may also be available inside the sub-packages. However, it is recommended to
+use the top-level compilation scripts to avoid forgetting to (re-)compile a sub-package's dependency.
+
+#### Run the dev mode
+
+Dev mode allows you to run the framework in the browser, using vue cli for fast development cycles, and easy debug tools.
+To run it do the following:
+
+- In the packages/backend folder run `webpack` or `webpack-dev:watch`, then run the server.
+  ```bash
+  yarn webpack-dev:watch
+  yarn ws:run
+  ```
+- In the packages/frontend folder run `serve`
+  ```bash
+  yarn serve
+  ```
+- Open the broswer on `localhost:8080` to access the framework.
+
+#### Run the VSCode extension
+
+- Start VSCode on your local machine, and click on open workspace. Select this repo folder.
+- On the debug panel choose "Run Extension", and click on the "Run" button.
+
+#### Enable usage analytics reporting from VS Code
+
+The tool collects non-personally identifiable information about your usage of the tool to improve its services. If you do not want the tool to collect your usage data, you can set the "Enable Sap Web Analytics" setting to "false". Go to File > Preferences > Settings (macOS: Code > Preferences > Settings) > Extensions > Application Wizard, and deselect the "Enable Sap Web Analytics" checkbox.
+
+#### Advanced scenarios
+
+- [Build & install the yeoman example generator](packages/generator-foodq/README.md)
+
+### Testing
+
+[Mocha][mocha] and [Chai][chai] are used for unit-testing and [Istanbul/Nyc][istanbul] for coverage reports for the TypeScript sub-packages and [Jest][jest] is used for unit-testing and coverage reports for the Vue sub-packages.
+
+[mocha]: https://mochajs.org/
+[chai]: https://www.chaijs.com
+[istanbul]: https://istanbul.js.org/
+[jest]: https://jestjs.io/
+
+- To run the tests execute `yarn test` in a specific sub-package.
+- To run the tests with **coverage** run `yarn coverage` in a specific sub-package.
+
+### Code Coverage
+
+Code Coverage is enforced for all productive code in this mono repo.
+
+- Specific statements/functions may be [excluded][ignore_coverage] from the report.
```

Comment:
are you using only jest? the documentation here links to istanbul ignore coverage docs, not jest...

Reply:
jest for frontend and nyc mocha for backend